deltachat-ffi/src/lib.rs

```diff
@@ -4785,6 +4786,91 @@ pub unsafe extern "C" fn dc_accounts_new(
     }
 }
 
+pub type dc_event_channel_t = Cell<Option<Events>>;
```

Comment:
`Cell` is not thread-safe. If C side uses this from multiple threads, this can result in memory corruption when trying to write into the same `Option` multiple times. Better use `Arc<Mutex<Option<Events>>>` so it is not possible to create two account managers by initializing them at the same time.

Reply:
> OTOH do we need thread safety here? We can just document that it's not thread-safe. 

All CFFI API is declared as thread-safe so we don't have to think about which is safe and which is not:
https://github.com/chatmail/core/blob/fd90493766311ae1648aa032738dff1c65dc75e4/deltachat-ffi/deltachat.h#L143-L147

It's probably not really true that all code is thread-safe because we don't have Mutexes around `Message` object, but that's a bug in that code, let's not add more bugs.